pkg/controller/kogitoruntime/deployer.go

```diff
@@ -73,13 +75,54 @@ func onGetComparators(comparator compare.ResourceComparator) {
 func onObjectsCreate(cli *client.Client, kogitoService v1alpha1.KogitoService) (resources map[reflect.Type][]resource.KubernetesResource, lists []runtime.Object, err error) {
 	resources = make(map[reflect.Type][]resource.KubernetesResource)
 
-	resObjectList, resType, res := createProtoBufConfigMap(kogitoService)
+	resObjectList, resType, res := createProtoBufConfigMap(cli, kogitoService)
 	lists = append(lists, resObjectList)
 	resources[resType] = []resource.KubernetesResource{res}
 	return
 }
 
-func createProtoBufConfigMap(kogitoService v1alpha1.KogitoService) (runtime.Object, reflect.Type, resource.KubernetesResource) {
+func getProtobufData(cli *client.Client, kogitoService v1alpha1.KogitoService) map[string]string {
+	available, err := services.IsDeploymentAvailable(cli, kogitoService)
+	if err != nil {
+		log.Errorf("failed to check status of %s, error message: %s", kogitoService.GetName(), err.Error())
+		return nil
+	}
+	if !available {
+		log.Debugf("deployment not available yet for %s ", kogitoService.GetName())
+		return nil
+	}
+
+	// print endpoint
+	protobufEndpoint := infrastructure.GetKogitoServiceEndpoint(kogitoService) + protobufSubdir
+	log.Debugf("%s Protobuf Endpoint: %s", kogitoService.GetName(), protobufEndpoint)
+
+	// print protobuf list
+	protobufListBytes, err := getHTTPFileBytes(protobufEndpoint + protobufListFileName)
```

Comment:
You need to check if the `http.StatusCode` is not 404, if it's the service does not have the persistence enabled, we shouldn't proceed. See the error:

```
{"level":"error","ts":1603750558.90423,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"kogitoruntime-controller","name":"dmn-drools-quarkus-metrics","namespace":"jenkins-cucumber-a612","error":"ConfigMap \"dmn-drools-quarkus-metrics-protobuf-files\" is invalid: data[RESTEASY003210: Could not find resource for full path: http://dmn-drools-quarkus-metrics.jenkins-cucumber-a612/persistence/protobuf/list.json]: Invalid value: \"RESTEASY003210: Could not find resource for full path: http://dmn-drools-quarkus-metrics.jenkins-cucumber-a612/persistence/protobuf/list.json\": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/jenkins/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/jenkins/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/jenkins/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/home/jenkins/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/jenkins/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/jenkins/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/jenkins/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/home/jenkins/go/pkg/mod/k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go:90"}
```

You only continue from here if the status is 200.

Reply:
Or even better, check in the `getHTTPFileBytes` and return an `nil` reference to the caller. That way you stop the execution if there's no content.

Reply:
Right, I meant to test an example without persistence. I'll add in this error checking and test it.